### ISSUE TITLE
Some questions about the paper

### ISSUE THREAD

**Issue by robin087** [Vance0124/Token-level-Direct-Preference-Optimization#7]
Thank you for this magnificent work!

I have several questions about the methodology of TDPO:
1. This paper directly transfers the derivation of DPO to TDPO, but why does the objective function contain only one token rather than the sum of all tokens?
2. For the preference function, the previous setting uses the optimal reward and policy, but your paper uses the reference ones. Does this change matter? 
3. In previous works about token-level DPO, like "From r to Q*" and "DPO meets PPO",  they conduct the loss function without the KL delta part in your paper. What is the difference between these two methods?

**Comment by Vance0124** [comment]
Regarding your first question, the reason our objective function contains only one token is related to the function itself. In our paper, our objective function (7) focuses on optimizing the expected future reward at each state (in this case, each token position), represented by the Advantage Function. This function inherently includes future information, as shown in equation (6) of the paper. Our goal is to maximize the expected future return at each state.


For your second question, this is tied to the proof in Appendix A.1 (Lemma A.1). Maximizing the Advantage Function relative to the reference policy enables us to obtain a new policy that is as optimal as possible. Theoretically, this necessitates the use of the reference policy, or the policy from the previous iteration. Additionally, from an experimental standpoint, reinforcement learning methods require a stable environment; using the reference policy throughout training process enhances the stability of our approach. We have also tested on-policy optimization updates based on the policy itself, but this often led to instability in the training environment of LLMs, resulting in training failures.


For your third question, I understand that "From r to Q*" and "DPO meets PPO" share a similar core idea, both deriving from the perspective of maximum entropy RL (i.e., Soft Q-Learning, SAC). In contrast, our method leans more toward derivation from TRPO and PPO.

Thank you for your understanding, and please feel free to reach out if you have any further questions.
